lib/std/Build/Step.zig

```diff
@@ -423,6 +423,8 @@ pub fn evalZigProcess(
         });
     }
 
+    if (s.cast(Compile)) |compile| if (compile.emit_bin == .no_emit) return result;
```

Comment:
Hmm can you extract this build stuff into a separate commit so I can understand better the intent of these changes? I think it conflicts with my recent changes in 6e4fff6ba62ae3e61a948c98fa8fea7e35732cc0 and my planned changes in #16351.

Reply:
See the `test:` commit.  If you want to try to find a different way to solve this, I would suggest only keeping the `test/` changes from that commit and solving the test failures you get from there.

Reply:
Thanks!

Reply:
Hmm, I think this is fine to merge as-is as long as it works. I can do my changes on top of this, no sweat.